android/ZBarScannerActivity.java

```diff
@@ -294,7 +294,7 @@ public void surfaceChanged (SurfaceHolder hld, int fmt, int w, int h)
 
         surfW = w;
         surfH = h;
-        matchSurfaceToPreviewRatio();
+        //matchSurfaceToPreviewRatio();
```

Comment:
I am a bit worried about commenting this.

The method has the following comment in it:
https://github.com/OutSystems/csZBar/blob/59bd5915a2d7f8f56c50b903db16cf754a2058b2/android/ZBarScannerActivity.java#L459-L460

What happens if the camera resolution has an aspect ratio that doesn't matches the aspect ratio from the screen? Reading from the comment I would say that we'll end up with a stretched camera preview, which isn't pretty.

I'd suggest attempting to recreate the surface on configuration change (screen rotation) so that the surface matches the aspect ratio of the screen on every moment in time.

Reply:
How about adding:
```
        tryStopPreview();
        tryStartPreview();
```
to the onSizeChanged callback while keeping the `matchSurfaceToPreviewRatio` calls?

Reply:
Hmm I tested this on a Pixel 3 XL where the screen has an aspect ratio of 16:9 whereas the camera is 4:3 and it still worked 🤔

Reply:
Followed @usernuno's suggestion, no joy. It only fixes one scenario, which is starting in portrait mode (with correct aspect ratio) and rotating to landscape once. Afterwards, rotating to portrait still distorts the preview and going back to landscape does nothing. Starting from landscape orientation still does not work (neither rotating back to portrait).

@Chuckytuh 

Correction: Pixel 3 XL has a screen aspect ratio of 18.5:9.

> I'd suggest attempting to recreate the surface on configuration change (screen rotation) so that the surface matches the aspect ratio of the screen on every moment in time.

I think that's what the plugin is attempting, but fails to do so.

I also tested on a Samsung Galaxy S7 where the screen has an aspect ration of 16:9 and the camera 4:3 and my approach works. Nasty, but effective. I have also seen other forks doings the same (or replacing the Android source code altogether but that's another story).